### PR TITLE
Port torch GPTQ to 3.x

### PR DESCRIPTION
**Pull request by yiliu30**
## Type of Change

Port torch GPTQ to 3.x

The 3rd PR for 3.X API
1. https://github.com/intel/neural-compressor/pull/1380
2. https://github.com/intel/neural-compressor/pull/1397

API changed:
- Add torch 3.x GPTQ API
~~Add `calib_dataloader` as a new arg for `quantize` as GPTQ needs a dataloader for calibration.~~
- To align with torch [quantize](https://github.com/pytorch/pytorch/blob/cd7d6938c18d90870356553d4631f1388d2bb699/torch/ao/quantization/quantize.py#L358), rename `calib_func` to `run_fn`, `calib_func_args` to `run_args`

## Description

- [x] Port GPTQ impl and add GPTQConfig
- [x] UTs for GPTQConfig
- [x] GPTQConfig + RTNConfig
- [x] Clean lwy import 

## How has this PR been tested?

Pre-CI

## Dependency Change?
None

### CODE REVIEW

File: neural_compressor/torch/quantization/quantize.py [hunk 1]
```diff
@@ -12,46 +12,57 @@
...
 
+def need_apply(configs_mapping: Dict[Tuple[str, callable], BaseConfig], algo_name):
+    return any(config.name == algo_name for config in configs_mapping.values())
+
+
 def quantize(
-    model: torch.nn.Module, quant_config: BaseConfig, calib_func: Callable = None, calib_func_arg: Any = None
+    model: torch.nn.Module,
+    quant_config: BaseConfig,
+    calib_dataloader=None,
```
Comment (yiliu30) [inline neural_compressor/torch/quantization/quantize.py:34]: Hi @ftian1, please be aware that I added `calib_dataloader` as a new argument for `quantize`, as GPTQ needs a dataloader for calibration.

Comment (ftian1) [reply neural_compressor/torch/quantization/quantize.py:34]: we need use run_fn / run_args to do that but not this calib_dataloader

**Review by YIYANGCAI — APPROVED** [review]

File: neural_compressor/torch/quantization/quantize.py (hunk 1 above)
Comment (yiliu30) [reply neural_compressor/torch/quantization/quantize.py]: removed and renamed by [7596137](https://github.com/intel/neural-compressor/pull/1408/commits/759613763b806f58edfd8ca6f4381ede69c0b505) and [1f02346](https://github.com/intel/neural-compressor/pull/1408/commits/1f02346ec24f94abfe44c3bc2050c0a5bc1afbfb)

Comment (yiliu30) [reply neural_compressor/torch/quantization/quantize.py]: refactored gptq to use `run_fn` and `run_args` by [42a07d9](https://github.com/intel/neural-compressor/pull/1408/commits/42a07d958fbd22efc8dea680f144b5a81d9f0c08)

**Pull request merged**